activestorage/lib/active_storage/attached.rb

```diff
@@ -16,6 +16,10 @@ def initialize(name, record)
       def change
         record.attachment_changes[name]
       end
+
+      def reset_changes
```

Comment:
This method sticks out to me, I wonder if we shouldn't be using [Attached::Changes](https://github.com/rails/rails/blob/d9749bebf64d26f9afb467ee7bad124ba38ad68d/activestorage/lib/active_storage/attached/changes.rb) interface instead? 🤔

Reply:
I understand your feeling, I'm just not sure how we could do it: we need to reset the changes which weren't being applied yet (because we are purging the attachment); how can make changes reset themselves?

In https://github.com/rails/rails/blob/d9749bebf64d26f9afb467ee7bad124ba38ad68d/activestorage/lib/active_storage/attached/model.rb#L196-L212 we have already some logic to manipulate attachments, you think adding this logic here is more consistent?

Reply:
@intrip Somehow I missed this, yeah I'm not sure the best place but I was just calling it out incase there was any better ideas. At least it's easy to refactor later as long as the API is private? 🙏

Reply:
@zzak Indeed: method visibility allows easy refactoring later on, if you agree I'd keep this change as it is for now.

Reply:
The `attachment_changes` api is definitely a weak spot - if you want to create a validator that checks an image before it's uploaded to S3 or whatever then you end up having to [reach deep inside some private apis][1] 😬 

I'd like to expose that in a better way but that's for a different PR - I think this is fine for the moment.

[1]: https://github.com/alphagov/e-petitions/blob/master/app/validators/image_validator.rb#L20-L39